Elmanager/LevelEditor/Shapes/CustomShapeControl.cs

```diff
@@ -0,0 +1,189 @@
+﻿using Elmanager.Lev;
+using Elmanager.Rendering;
+using OpenTK.GLControl;
+using System;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using Color = System.Drawing.Color;
+using Pen = System.Drawing.Pen;
+
+namespace Elmanager.LevelEditor.Shapes;
+
+internal partial class CustomShapeControl : UserControl
+{
+    private bool _isHighlighted;
+    private bool _isPressed;
+    private bool _isSelected;
+
+    private Color _borderColor = Color.Transparent;
+
+    public CustomShapeControl(GLControl sharedContext, SceneSettings sceneSettings, RenderingSettings renderingSettings, ElmaRenderer elmaRenderer, Level level)
+    {
+        InitializeComponent();
+
+        // Filter start object from level
+        level.Objects = level.Objects.Where(o => o.Type != ObjectType.Start).ToList();
+
+        shapeLevelControl = new LevelControl(sharedContext, sceneSettings, renderingSettings, elmaRenderer, level);
+        shapeLevelControl.API = OpenTK.Windowing.Common.ContextAPI.OpenGL;
+        shapeLevelControl.APIVersion = new Version(3, 3, 0, 0);
+        shapeLevelControl.Flags = OpenTK.Windowing.Common.ContextFlags.Default;
+        shapeLevelControl.Location = new System.Drawing.Point(0, 0);
+        shapeLevelControl.Size = new Size(102, 102);
+        shapeLevelControl.Margin = new Padding(0);
+        shapeLevelControl.Padding = new Padding(0);
+        shapeLevelControl.Name = "shapeLevelControl";
+        shapeLevelControl.Dock = DockStyle.None;
+        shapeLevelControl.Profile = OpenTK.Windowing.Common.ContextProfile.Compatability;
+        shapeLevelControl.TabIndex = 0;
+        shapeLevelControl.TabStop = false;
+        tableLayoutPanel1.Controls.Add(shapeLevelControl, 0, 0);
+
+        shapeLevelControl.Click += OnComponentClick;
+        lblShapeName.Click += OnComponentClick;
+        this.Click += OnComponentClick;
+
+        shapeLevelControl.MouseEnter += OnMouseEnter;
+        lblShapeName.MouseEnter += OnMouseEnter;
+        this.MouseEnter += OnMouseEnter;
+
+        shapeLevelControl.MouseLeave += OnMouseLeave;
+        lblShapeName.MouseLeave += OnMouseLeave;
+        this.MouseLeave += OnMouseLeave;
+
+        shapeLevelControl.MouseDown += OnMouseDown;
+        lblShapeName.MouseDown += OnMouseDown;
+        this.MouseDown += OnMouseDown;
+
+        shapeLevelControl.MouseUp += OnMouseUp;
+        lblShapeName.MouseUp += OnMouseUp;
+        this.MouseUp += OnMouseUp;
+
+        shapeLevelControl.DoubleClick += CustomShapeControl_DoubleClick;
+        lblShapeName.DoubleClick += CustomShapeControl_DoubleClick;
+        this.DoubleClick += CustomShapeControl_DoubleClick;
+    }
+
+    public string ShapeFullPath { get; set; } = String.Empty;
```

Comment:
I'd rather make this nullable instead of using magic invalid value:

```suggestion
    public string? ShapeFullPath { get; set; };
```

Reply:
Fixed in c50fec9d24f45188b8d89ecb9e015aa826370299